src/gf.c

```diff
@@ -2942,6 +2942,19 @@ JL_DLLEXPORT int jl_compile_hint(jl_tupletype_t *types)
     return 1;
 }
 
+// true if jl_get_compile_hint_specialization can find a method instance for us to try
+// to compile
+JL_DLLEXPORT int jl_has_compile_hint_specialization(jl_tupletype_t *types)
```

Comment:
this is just `jl_compile_hint` from above (which is what `precompile(f, argtypes)` calls, but without the `jl_compile_method_instance` call